spec/forms/support/provider_form_spec.rb

```diff
@@ -0,0 +1,178 @@
+
+      include_examples 'blank validation', :accredited_provider_id, 'Enter an accredited provider ID'
+    end
+
+    context 'urn set to invalid' do
+      let(:params) do
+        {
+          provider_type: :lead_school,
+          urn: %w[1234 1234567 digit].sample
+        }
+      end
+
+      it 'validates the urn' do
+        expect(subject).not_to be_valid
+
+        expect(subject.errors[:urn]).to match_array('Enter a valid unique reference number (URN)')
+      end
+    end
+
+    context 'ukprn set to invalid' do
+      let(:params) do
+        {
+          ukprn: %w[1234567 123456789 digit].sample
+        }
+      end
+
+      it 'validates the ukprn' do
+        expect(subject).not_to be_valid
+
+        expect(subject.errors[:ukprn]).to match_array('Enter a valid UK provider reference number (UKPRN)')
+      end
+    end
+
+    context 'using an existing provider_code' do
+      let(:params) do
+        { provider_code: }
+      end
+
+      let(:provider_code) do
+        create(:provider).provider_code
+      end
+
+      it 'validates the provider code' do
+        expect(subject).not_to be_valid
+
+        expect(subject.errors[:provider_code]).to match_array('Provider code already taken')
+      end
+    end
+
+    context 'provider_type is set to lead_school and accredited_provider is set to accredited_body' do
+      let(:params) do
+        {
+          provider_type: :lead_school,
+          accredited_provider: :accredited_body
+        }
+      end
+
+      it 'validates the provider type' do
+        expect(subject).not_to be_valid
+
+        expect(subject.errors[:provider_type]).to match_array('Accredited provider cannot be a school')
+      end
+    end
+
+    shared_examples 'accredited provider id validation' do |provider_type, accredited_provider_id, message|
+      context "provider_type is set to '#{provider_type}' and accredited_provider_id is set to '#{accredited_provider_id}'" do
+        let(:params) do
+          {
+            provider_type:,
+            accredited_provider: :accredited_body,
+            accredited_provider_id:
+          }
+        end
+
+        it 'validates the accredited provider id' do
+          expect(subject).not_to be_valid
+
+          expect(subject.errors[:accredited_provider_id]).to match_array(message)
+        end
+      end
+    end
+
+    blank_accredited_provider_id_message = 'Enter an accredited provider ID'
+
+    [nil, ''].each do |blank_accredited_provider_id|
+      include_examples 'accredited provider id validation', nil, blank_accredited_provider_id, blank_accredited_provider_id_message
+      include_examples 'accredited provider id validation', :scitt, blank_accredited_provider_id, blank_accredited_provider_id_message
+      include_examples 'accredited provider id validation', :university, blank_accredited_provider_id, blank_accredited_provider_id_message
+    end
+
+    invalid_accredited_provider_id_message = 'Enter a valid accredited provider ID'
```

Comment:
Could we have a space here similar to above

Reply:
Done